lib/honeydew.ex

```diff
@@ -35,6 +64,12 @@ defmodule Honeydew do
     job
   end
 
+  @doc """
+  Wait for a job to complete and return result.
+
+  Returns the result of a job, or `nil` on timeout.
+  """
```

Comment:
can you add a bit about how `async/2` can be called multiple times, and must be called from the enqueueing process?

Reply:
What do you mean by this? Wouldn't calling `async/2` multiple times enqueue another job?

Reply:
ugh, sorry, i meant `yield/2`.

Reply:
So you can do something like this?

``` elixir
iex> job = Honeydew.async({:ping, ["127.0.0.1"], :my_queue, reply: true)
iex> Enum.map(1..5, &Honeydew.yield/1)
[:pong, :pong, :pong, :pong, :pong]
```
or 
``` elixir
iex> job = Honeydew.async({:ping, ["127.0.0.1"], :my_queue, reply: true)
iex> Honeydew.yield(job, 2000)
nil
# Result comes in at 3 seconds
iex> Honeydew.yield(job, 5000)
:pong
iex> Honeydew.yield(job, 10000)
:pong
```

Reply:
almost:

```
iex> job = Honeydew.async({:ping, ["127.0.0.1"], :my_queue, reply: true)
iex> Honeydew.yield(job, 2000)
nil
# Result comes in at 3 seconds
iex> Honeydew.yield(job, 5000)
:pong
iex> Honeydew.yield(job, 10000)
nil # <-- because the message has already arrived and been handled
```

the only time that `yield/2` would ever return `:pong` more than once is if the job executes more than once (as honeydew aims for at-least-once)